OnlineFeedViewActivity: SQLiteDatabaseLockedException

App version: 1.6.1.2

Not sure if this is even related to said activity. Might be that the db is busy writing data or is in a long-running transaction causing it to throw is locked.

We can at least fix the crashing part by adding onError...

```
java.lang.IllegalStateException: Exception thrown on Scheduler.Worker thread. Add `onError` handling.
	at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:60)
	at android.os.Handler.handleCallback(Handler.java:605)
	at android.os.Handler.dispatchMessage(Handler.java:92)
	at android.os.Looper.loop(Looper.java:137)
	at android.app.ActivityThread.main(ActivityThread.java:4424)
	at java.lang.reflect.Method.invokeNative(Native Method)
	at java.lang.reflect.Method.invoke(Method.java:511)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:791)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:558)
	at dalvik.system.NativeStart.main(Native Method)
Caused by: rx.exceptions.OnErrorNotImplementedException: database is locked
	at rx.Observable$27.onError(Observable.java:7923)
	at rx.observers.SafeSubscriber.onError(SafeSubscriber.java:2159)
	at rx.internal.operators.OperatorObserveOn$ObserveOnSubscriber.pollQueue(OperatorObserveOn.java:191)
	at rx.internal.operators.OperatorObserveOn$ObserveOnSubscriber$2.call(OperatorObserveOn.java:162)
	at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:55)
	... 9 more
Caused by: android.database.sqlite.SQLiteDatabaseLockedException: database is locked
	at android.database.sqlite.SQLiteDatabase.native_setLocale(Native Method)
	at android.database.sqlite.SQLiteDatabase.setLocale(SQLiteDatabase.java:2074)
	at android.database.sqlite.SQLiteDatabase.openDatabase(SQLiteDatabase.java:1014)
	at android.database.sqlite.SQLiteDatabase.createPoolConnection(SQLiteDatabase.java:2393)
	at android.database.sqlite.DatabaseConnectionPool.get(DatabaseConnectionPool.java:114)
	at android.database.sqlite.SQLiteDatabase.getDbConnection(SQLiteDatabase.java:2427)
	at android.database.sqlite.SQLiteDatabase.rawQueryWithFactory(SQLiteDatabase.java:1559)
	at android.database.sqlite.SQLiteDatabase.queryWithFactory(SQLiteDatabase.java:1449)
	at android.database.sqlite.SQLiteDatabase.query(SQLiteDatabase.java:1405)
	at android.database.sqlite.SQLiteDatabase.query(SQLiteDatabase.java:1485)
	at de.danoeh.antennapod.core.storage.PodDBAdapter.getAllFeedsCursor(PodDBAdapter.java:1071)
	at android.support.design.R.getFeedList(R.java:51291)
	at android.support.design.R.getFeedList(R.java:51288)
	at de.danoeh.antennapod.activity.OnlineFeedViewActivity$2.call(OnlineFeedViewActivity.java:1274)
	at rx.Observable.unsafeSubscribe(Observable.java:8098)
	at rx.internal.operators.OperatorSubscribeOn$1$1.call(OperatorSubscribeOn.java:62)
	at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:55)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:442)
	at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:305)
	at java.util.concurrent.FutureTask.run(FutureTask.java:137)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:150)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:264)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1076)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:569)
	at java.lang.Thread.run(Thread.java:856)
```

--- a/app/src/main/java/de/danoeh/antennapod/activity/OnlineFeedViewActivity.java
+++ b/app/src/main/java/de/danoeh/antennapod/activity/OnlineFeedViewActivity.java
@@ -119,9 +119,12 @@ public class OnlineFeedViewActivity extends AppCompatActivity {
                 updater = Observable.fromCallable(() -> DBReader.getFeedList())
                         .subscribeOn(Schedulers.newThread())
                         .observeOn(AndroidSchedulers.mainThread())
-                        .subscribe(feeds -> {
+                        .subscribe(
+                                feeds -> {
                                     OnlineFeedViewActivity.this.feeds = feeds;
                                     setSubscribeButtonState(feed);
+                                }, error -> {
+                                    Log.e(TAG, Log.getStackTraceString(error));
                                 }
                         );
             } else if ((arg & EVENTS) != 0) {
@@ -281,30 +284,33 @@ public class OnlineFeedViewActivity extends AppCompatActivity {
                 })
                 .subscribeOn(Schedulers.newThread())
                 .observeOn(AndroidSchedulers.mainThread())
-                .subscribe(status -> {
-                    if (status != null) {
-                        if (!status.isCancelled()) {
-                            if (status.isSuccessful()) {
-                                parseFeed();
-                            } else if (status.getReason() == DownloadError.ERROR_UNAUTHORIZED) {
-                                if (!isFinishing() && !isPaused) {
-                                    dialog = new FeedViewAuthenticationDialog(OnlineFeedViewActivity.this,
-                                            R.string.authentication_notification_title, downloader.getDownloadRequest().getSource());
-                                    dialog.show();
-                                }
-                            } else {
-                                String errorMsg = status.getReason().getErrorString(OnlineFeedViewActivity.this);
-                                if (errorMsg != null && status.getReasonDetailed() != null) {
-                                    errorMsg += " (" + status.getReasonDetailed() + ")";
-                                }
-                                showErrorDialog(errorMsg);
-                            }
-                        }
-                    } else {
-                        Log.wtf(TAG, "DownloadStatus returned by Downloader was null");
-                        finish();
-                    }
-                });
+                .subscribe(status -> checkDownloadResult(status),
+                        error -> Log.e(TAG, Log.getStackTraceString(error)));
+    }
+
+    private void checkDownloadResult(DownloadStatus status) {
+        if (status == null) {
+            Log.wtf(TAG, "DownloadStatus returned by Downloader was null");
+            finish();
+        }
+        if (status.isCancelled()) {
+            return;
+        }
+        if (status.isSuccessful()) {
+            parseFeed();
+        } else if (status.getReason() == DownloadError.ERROR_UNAUTHORIZED) {
+            if (!isFinishing() && !isPaused) {
+                dialog = new FeedViewAuthenticationDialog(OnlineFeedViewActivity.this,
+                        R.string.authentication_notification_title, downloader.getDownloadRequest().getSource());
+                dialog.show();
+            }
+        } else {
+            String errorMsg = status.getReason().getErrorString(OnlineFeedViewActivity.this);
+            if (errorMsg != null && status.getReasonDetailed() != null) {
+                errorMsg += " (" + status.getReasonDetailed() + ")";
+            }
+            showErrorDialog(errorMsg);
+        }
     }
 
     private void parseFeed() {

